rust/connlib/ip-packet/src/lib.rs

```diff
@@ -354,6 +354,20 @@ impl IpPacket {
         ))
     }
 
+    pub fn layer4_payload_len(&self) -> usize {
+        let tcp_payload_len = self.as_tcp().map(|tcp| tcp.payload().len());
+        let udp_payload_len = self.as_udp().map(|udp| udp.payload().len());
+        let icmpv4_payload_len = self.as_icmpv4().map(|icmpv4| icmpv4.payload().len());
+        let icmpv6_payload_len = self.as_icmpv6().map(|icmpv6| icmpv6.payload().len());
+        let fallback = self.payload().len();
+
+        tcp_payload_len
+            .or(udp_payload_len)
+            .or(icmpv4_payload_len)
+            .or(icmpv6_payload_len)
+            .unwrap_or(fallback)
+    }
+
```

Comment:
could we  do lazy evaluation here? we only return one value

```suggestion
pub fn layer4_payload_len(&self) -> usize {
    if let Some(tcp) = self.as_tcp() {
        return tcp.payload().len();
    }
    if let Some(udp) = self.as_udp() {
        return udp.payload().len();
    }
    if let Some(icmpv4) = self.as_icmpv4() {
        return icmpv4.payload().len();
    }
    if let Some(icmpv6) = self.as_icmpv6() {
        return icmpv6.payload().len();
    }
    self.payload().len()
}
```

could also do the functional / concise way, by inlining all these variables

Reply:
It is in the hot-path so might be worth optimising.